src/games/strategy/triplea/attachments/TerritoryAttachment.java

```diff
@@ -138,15 +138,7 @@ public static Territory getFirstOwnedCapitalOrFirstUnownedCapital(final PlayerID
    * Convenience method. Can return null.
    */
   public static TerritoryAttachment get(final Territory t) {
-    return get(t, false);
-  }
-
-  public static TerritoryAttachment get(final Territory t, final boolean allowNull) {
-    final TerritoryAttachment rVal = (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
-    if (!allowNull && rVal == null && !t.isWater()) {
-      throw new IllegalStateException("No territory attachment for:" + t.getName());
-    }
-    return rVal;
+    return (TerritoryAttachment) t.getAttachment(Constants.TERRITORY_ATTACHMENT_NAME);
```

Comment:
Perhaps, though I did not find any and am relatively sure they do not exist (or at least of any consequence). Yes we may lose some accuracy in the error message, though we can fill that detail in when we run across it. 
 I checked out usages of this function, there are not *that* many, we pretty consistently account for null possibly coming back. It really seems the throwing by default behavior is not correct/intended. It seems like this method was more used because it was there, rather than used with thought that we throw on no territory attachment found. Basically had the code been designed with the throwing behavior in mind, we wouldn't see all the null handling on the return of this method.
Bottom line and the point is that we gain in simplicity by not throwing, an easier test setup and just some less code, and it does not seem we are losing much of anything (and are certainly not introducing any new errors)